### PR TITLE
util: add from_encoded_slice

### PR DESCRIPTION
**Pull request by Connor1996**
Signed-off-by: Connor1996 <zbk602423539@gmail.com>

## What have you changed? (mandatory)
- add `from_encoded_slice()` that reserve capacity for timestamp
- call `from_encoded_slice()` instead of `from_encoded(xxx.to_vec()/to_own())` in some code place, but not all cause `xxx` maybe timestamped key slice and there is no need to reserve more capacity.

## What are the type of the changes? (mandatory)
- Improvement (non-breaking change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)
No need

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)
No need

## Does this PR affect tidb-ansible update? (mandatory)
No need

### CODE REVIEW

File: src/storage/types.rs [hunk 1]
```diff
@@ -71,12 +71,20 @@ impl Key {
...
     }
 
-    /// Creates a key from encoded bytes.
+    /// Creates a key from encoded bytes vector.
     #[inline]
     pub fn from_encoded(encoded_key: Vec<u8>) -> Key {
         Key(encoded_key)
     }
 
+    /// Creates a key with reserver capacity for timestamp from encoded bytes slice.
```
Comment (MyonKeminta) [inline src/storage/types.rs]: /s/reserver/reserved

**Comment by Connor1996** [comment]
Please approve again @breeswish @MyonKeminta 😅

**Comment by breeswish** [comment]
@MyonKeminta Please help utilize this function in our mvcc refinement.

**Pull request merged**